### PR TITLE
Enable most checks in pylint

### PR DESCRIPTION
**Pull request by ldoktor**
OK, we want to cleanup Avocado, let's start with a smaller project first. There are some probably controversial commits so I'm open to cherry-picking and hammering out the details in following PRs.

### CODE REVIEW

**Comment by ldoktor** [comment]
Just for record:
```
./aexpect/exceptions.py: Your code has been rated at 10.00/10 (previous run: 7.78/10, +2.22)
./aexpect/utils/astring.py: Your code has been rated at 10.00/10 (previous run: 9.70/10, +0.30)
./aexpect/utils/wait.py: Your code has been rated at 10.00/10 (previous run: 8.67/10, +1.33)
./aexpect/utils/process.py: Your code has been rated at 10.00/10 (previous run: 9.20/10, +0.80)
./aexpect/utils/__init__.py:
./aexpect/utils/data_factory.py: Your code has been rated at 10.00/10 (previous run: 9.41/10, +0.59)
./aexpect/utils/path.py: Your code has been rated at 10.00/10 (previous run: 9.29/10, +0.71)
./aexpect/__init__.py: Your code has been rated at 10.00/10 (previous run: 9.38/10, +0.62)
./aexpect/client.py: Your code has been rated at 10.00/10 (previous run: 7.88/10, +2.12)
./aexpect/shared.py: Your code has been rated at 10.00/10 (previous run: 7.50/10, +2.50)
./tests/test_pass_fds.py: Your code has been rated at 10.00/10 (previous run: 8.33/10, +1.67)
./tests/test_client.py: Your code has been rated at 10.00/10 (previous run: 8.67/10, +1.33)
./tests/__init__.py:
./setup.py: Your code has been rated at 10.00/10 (previous run: 8.57/10, +1.43)
```

**Comment by ldoktor** [comment]
OK the "set_log_file" is actually being used in Avocado-vt in a nasty way... I'll have to keep that for the compatibility reasons...

**Comment by lmr** [comment]
Let me start

File: aexpect/client.py [hunk 1]
```diff
@@ -1130,14 +1130,15 @@ def cmd_output(self, cmd, timeout=60, internal_timeout=None,
...
-                raise ShellError(cmd, o)
+        except ExpectTimeoutError as error:
+            output = self.remove_command_echo(error.output, cmd)
+            raise ShellTimeoutError(cmd, output)
+        except ExpectProcessTerminatedError as error:
+            output = self.remove_command_echo(error.output, cmd)
+            raise ShellProcessTerminatedError(cmd, error.status, output)
+        except ExpectError as error:
+            output = self.remove_command_echo(error.output, cmd)
+            raise ShellError(cmd, output)
```
Comment (lmr) [inline aexpect/client.py]: Sure, the only small problem is that we have to repeat a bit the 'output = ' line.

File: Makefile [hunk 2]
```diff
@@ -73,7 +73,7 @@ rpm-release: srpm-release
 	mock -r $(MOCK_CONFIG) --resultdir BUILD/RPM -D "rel_build 1" --rebuild BUILD/SRPM/python-aexpect-$(VERSION)-*.src.rpm
 
 check: clean
-	inspekt checkall --disable-lint W1618,R,C,E1002,E1101,E1103,E1120,F0401,I0011
+	inspekt checkall --disable-lint W1618,R,C
```
Comment (lmr) [inline Makefile]: great

File: aexpect/utils/genio.py [hunk 3]
```diff
@@ -9,18 +9,29 @@
 #
 # See LICENSE for more details.
 
+"""
+Naive module that keeps tacks of some opened files and somehow manages them.
+"""
```
Comment (lmr) [inline aexpect/utils/genio.py]: Yep, I agree that this should go eventually. But first things first

File: aexpect/client.py (hunk 1 above)
Comment (ldoktor) [reply aexpect/client.py]: yes, but by disabling the check we can hide missing attributes (which actually happened to me while addressing this due to minor mistake I did. That's why I chose this way :smile:

File: aexpect/utils/genio.py (hunk 3 above)
Comment (ldoktor) [reply aexpect/utils/genio.py]: Yep, I wanted to turn it into a public interface or remove, but when I saw how is it used in Avocado-vt I just created: https://trello.com/c/6zsHUXYr/1351-aexpect-get-rid-of-aexpectutilsgenio

**Pull request merged**